src/plugins/data/common/search/aggs/utils/calculate_auto_time_expression.ts

```diff
@@ -8,14 +8,36 @@
  */
 
 import moment from 'moment';
+import { TimeBucketsInterval } from '../buckets/lib/time_buckets/time_buckets';
 import { UI_SETTINGS } from '../../../constants';
 import { TimeRange } from '../../../query';
 import { TimeBuckets } from '../buckets/lib/time_buckets';
 import { toAbsoluteDates } from './date_interval_utils';
 import { autoInterval } from '../buckets/_interval_options';
 
 export function getCalculateAutoTimeExpression(getConfig: (key: string) => any) {
-  return function calculateAutoTimeExpression(range: TimeRange) {
+  function calculateAutoTimeExpression(range: TimeRange): string | undefined;
+  function calculateAutoTimeExpression(
+    range: TimeRange,
+    interval: string,
+    expression?: true
+  ): string | undefined;
+  function calculateAutoTimeExpression(
+    range: TimeRange,
+    interval: string,
+    expression: false
+  ): TimeBucketsInterval | undefined;
+  function calculateAutoTimeExpression(
+    range: TimeRange,
+    interval?: string,
+    expression?: boolean
+  ): string | TimeBucketsInterval | undefined;
+
+  function calculateAutoTimeExpression(
+    range: TimeRange,
+    interval: string = autoInterval,
+    expression: boolean = true
```

Comment:
For readability, can this argument name be changed to something like `asExpression` so reads as a boolean?